src/cfnlint/rules/resources/iam/InstanceProfile.py

```diff
@@ -53,7 +53,7 @@ def match(self, cfn):
                                 message = 'Property IamInstanceProfile should relate to AWS::IAM::InstanceProfile for %s' % (
                                     '/'.join(map(str, tree[:-1])))
                                 matches.append(RuleMatch(tree[:-1], message))
-                        elif cfn.template.get('Resources', {}).get(tree[1], {}).get('Type') in ['AWS::EC2::SpotFleet']:
+                        elif cfn.template.get('Resources', {}).get(tree[1], {}).get('Type') in ['AWS::EC2::SpotFleet', 'AWS::EC2::LaunchTemplate']:
                             if obj[1] != 'Arn':
```

Comment:
So, looking at the docs (https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-launchtemplate-iaminstanceprofile.html) it appears you can actually specify either an IamInstanceProfile Arn or a Name.  

This may need to be its own block, since AWS::EC2::SpotFleet only accepts Arns.

Reply:
I've extended the statement to give LaunchTemplate it's own rule.